### ISSUE TITLE
maracluster consensus produced more consensus spectra than expected

### ISSUE THREAD

**Issue by JannikSchneider12** [statisticalbiotechnology/maracluster#29]
Hey,

I have a problem with the consensus spectra generation. So I used the output of maracluster batch, filtered the file based on my criteria and write back a tsv file with only my remaining specta where I want to create the consensus spectra. I gave this file to the maracluster consensus function but the resulting file (in my case mgf) contains more spectra than I would expect.

So this is my code for reading in the input files, filter based on my criteria and write the resulting tsv files back:

![image](https://github.com/statisticalbiotechnology/maracluster/assets/72038436/c8237f9b-5c96-44e2-8991-137fe718885d)

And here are the lenghts differences:

![image](https://github.com/statisticalbiotechnology/maracluster/assets/72038436/72886aad-284e-4bdc-9718-85900274e4c8)

Moreover I would have another question. So the title of the consensus spectra is just "scan=XXX" but no parameter for the cluster index. So is the value for "scan=" the cluster index?

Thanks for your time and help

**Comment by MatthewThe** [comment]
One thing that could play a role here is that the consensus mgf output will make several copies for each consensus spectrum for each of the precursors it could correspond to. This is because the mgf format does not allow specifying multiple precursors per scan. This usually only happens if the input spectrum file was peak picked, so I'm not sure if it applies here.

The scan number of consensus spectra is documented here: https://github.com/statisticalbiotechnology/maracluster/wiki/FAQ

**Comment by JannikSchneider12** [comment]
Hey,

thank you for the explanation. And yes I converted the raw files via peak picking. Is there a way to avoid the creation of multiple consensus spectra for the same spectrum or does it make sense to do so, since there could still be different precursor charges although it is not „shown“ in the mgf? (Sorry I am new to mass spectrometry)

thanks again for your time and help

**Comment by MatthewThe** [comment]
Sorry, I misspoke. I actually meant precursor detection rather than peak picking. Some search engines (e.g. MaxQuant and FragPipe) do this automatically, but many others don't. For search engines that don't do this (e.g. Comet and MSGF+), performing precursor detection (e.g. with Bullseye or Dinosaur/Biosaur) prior to clustering+searching can drastically increase the identification rate.

Can you send me the first 100 scan numbers of your consensus mgf, that would quickly tell me if we indeed have multiple precursors per spectrum.

**Comment by JannikSchneider12** [comment]
Here are the first scan numbers of the maracluster output:

![image](https://github.com/statisticalbiotechnology/maracluster/assets/72038436/dc9ef5f8-8196-4513-abf0-8cd1cea83a91)

**Comment by MatthewThe** [comment]
Yes, you indeed have multiple precursors per consensus spectrum as some scan numbers end with 02. 

If you're fine with using a different spectrum format, e.g. .mzML or .ms2, you can get a single spectrum (with multiple precursors) per consensus spectrum.

**Comment by JannikSchneider12** [comment]
Thanks again. Just for curiosity but since MaRaCluster gets in my case just the mgf files, I was asking myself where it gets the information from that there might be multiple precursors? And did I get it right that in those cases where multiple consensus spectra are created for the same cluster, then they just differ by the precursor charge?

**Comment by MatthewThe** [comment]
It's hard to say without seeing the input files. The most likely reason is indeed that some spectra get assigned multiple potential charge states, either by the mass spec or by a preprocessing step. This then would indeed lead to the corresponding consensus spectra having multiple potential charge states.

**Comment by JannikSchneider12** [comment]
Alright, then I will try to have a deeper look. Thanks again for your help!
